awx/ui_next/src/screens/Template/WorkflowJobTemplateVisualizer/Visualizer.jsx

```diff
@@ -187,19 +797,125 @@ function Visualizer({ template, i18n }) {
+            onLinkEditClick={setLinkToEdit}
+            onStartAddLinkClick={selectSourceNodeForLinking}
+            onUpdateShowKey={setShowKey}
+            onUpdateShowTools={setShowTools}
+            onViewNodeClick={setNodeToView}
+            readOnly={!template.summary_fields.user_capabilities.edit}
+            showKey={showKey}
+            showTools={showTools}
+          />
+        ) : (
+          <VisualizerStartScreen onStartClick={startAddNode} />
+        )}
+      </Wrapper>
+      {nodeToDelete && (
+        <NodeDeleteModal
+          nodeToDelete={nodeToDelete}
+          onCancel={() => setNodeToDelete(null)}
+          onConfirm={deleteNode}
+        />
+      )}
+      {linkToDelete && (
+        <LinkDeleteModal
+          linkToDelete={linkToDelete}
+          onCancel={() => setLinkToDelete(null)}
+          onConfirm={deleteLink}
+        />
+      )}
+      {linkToEdit && (
+        <LinkModal
+          linkType={linkToEdit.linkType}
+          header={
+            <Title headingLevel={TitleLevel.h1} size={BaseSizes['2xl']}>
+              {i18n._(t`Edit Link`)}
+            </Title>
+          }
+          onCancel={() => setLinkToEdit(null)}
+          onConfirm={updateLink}
+        />
+      )}
+      {addLinkSourceNode && addLinkTargetNode && (
+        <LinkModal
+          header={
+            <Title headingLevel={TitleLevel.h1} size={BaseSizes['2xl']}>
+              {i18n._(t`Add Link`)}
+            </Title>
+          }
+          onCancel={cancelNodeLink}
+          onConfirm={addLink}
+        />
+      )}
+      {addNodeSource && (
+        <NodeModal
+          askLinkType={addNodeSource !== 1}
+          onClose={() => cancelNodeForm()}
+          onSave={finishAddingNode}
+          title={i18n._(t`Add Node`)}
         />
-      ) : (
-        <StartScreen />
       )}
-    </VisualizerLayout>
+      {nodeToEdit && (
+        <NodeModal
+          askLinkType={false}
+          nodeToEdit={nodeToEdit}
+          onClose={() => cancelNodeForm()}
+          onSave={finishEditingNode}
+          title={i18n._(t`Edit Node`)}
+        />
+      )}
+      {showUnsavedChangesModal && (
+        <UnsavedChangesModal
+          onCancel={() => setShowUnsavedChangesModal(false)}
+          onExit={() =>
+            history.push(
+              `/templates/workflow_job_template/${template.id}/details`
+            )
+          }
+          onSaveAndExit={() => handleVisualizerSave()}
+        />
+      )}
+      {showDeleteAllNodesModal && (
+        <DeleteAllNodesModal
+          onCancel={() => setShowDeleteAllNodesModal(false)}
+          onConfirm={() => deleteAllNodes()}
+        />
+      )}
+      {nodeToView && (
+        <NodeViewModal node={nodeToView} onClose={() => setNodeToView(null)} />
+      )}
+    </Fragment>
   );
 }
 
-export default withI18n()(Visualizer);
+Visualizer.propTypes = {
+  template: shape().isRequired,
+};
+
+export default withI18n()(withRouter(Visualizer));
```

Comment:
There seems to be a lot of duplication between WorkflowOutput/WorkflowOutputGraph and Visualizer/VisualizerGraph, as well as the link/node/start node/start screen, toolbar and help  In an ideal world, it would be nice if you could use the same backbone and put the "form-on-click" kind of stuff in here.  Not something I think we need to change for this PR, but the duplication does seem like it may cause change drift between view and edit over time.